narayana/ArjunaJTA/jta/tests/classes/org/jboss/narayana/rts/EmptyTxnTest.java

```diff
@@ -50,14 +48,14 @@ public static void tearDown() throws IOException {
 
     @Benchmark
     public void testEmptyTxn(Blackhole bh) throws IOException {
-        bh.consume(beginTx());
-        // make two service calls for comparison with RTSTests#testTxn
-        bh.consume(sendRequest());
-        bh.consume(TxnHelper.endTxn(txnClient, links));
+        bh.consume(testEmptyTxn());
     }
 
-    public boolean beginTx() throws IOException {
-        links = TxnHelper.beginTxn(txnClient, TXN_URL);
+    public boolean testEmptyTxn() throws IOException {
+        Set<Link> links = TxnHelper.beginTxn(txnClient, TXN_URL);
+        // make two service calls for comparison with RTSTests#testTxn
+        sendRequest();
+        TxnHelper.endTxn(txnClient, links);
```

Comment:
With these changes we are no longer consuming the output of starting and ending the transaction and the send request, are you sure we don't need to protect against dead-code elimination?

Reply:
This is a very good question! I might be wrong, but I think calling the Blackhole.consume() method at the beginning (https://github.com/jbosstm/performance/pull/155/files#diff-ab204da3be84079af8c97fbf0b5837ab131083cc2d51bc71d1272653ffc8c95fR51) is enough to prevent against dead call elimination. I will double check it and reference here what I find.

Reply:
I believe the following would ensure that JMH does not eliminate any dead code:

```
    public boolean testEmptyTxn() throws IOException {
        Set<Link> links = TxnHelper.beginTxn(txnClient, TXN_URL);
        // make two service calls for comparison with RTSTests#testTxn
        boolean sent = sendRequest();
        int status = TxnHelper.endTxn(txnClient, links);
        
        return (status == TxStatus.TransactionCommitted.ordinal() && sent);
    }
```